src/lib/matcher.rs

```diff
@@ -407,3 +408,252 @@ impl MatcherFactory {
+    #[case("A.A", "ATATA".as_bytes(), bitvec![1, 1, 1, 0, 0])] // regex with overlapping matches
+    fn test_regex_bitvec(#[case] pattern: &str, #[case] bases: &[u8], #[case] expected: BitVec) {
+        let opts = MatcherOpts {
+            invert_match: false,
+            reverse_complement: false,
+            color: false,
+        };
+        let matcher = RegexMatcher::new(&pattern, opts);
+        let ranges = matcher.regex.find_iter(bases).map(|m| m.range());
+        let result_bitvec = to_bitvec(ranges, bases.len());
+        println!("{:?}", &result_bitvec);
+        assert_eq!(result_bitvec, expected);
+    }
+
+    // ############################################################################################
+    // Tests to_bitvec() regex set
+    // ############################################################################################
+    #[rstest]
+    #[case(vec!["^A"], "AGG".as_bytes(), bitvec![1, 0, 0])] // regex set with one pattern and one match
+    #[case(vec!["^A"], "CGG".as_bytes(), bitvec![0, 0, 0])] // regex set with one pattern and no matches
+    #[case(vec!["^A.G", "C..", "$T"], "AGGCTT".as_bytes(), bitvec![1, 1, 1, 1, 1, 1])] // regex set with multiple patterns that all match
+    #[case(vec!["^A.G", "C..", "$T"], "TTTTTC".as_bytes(), bitvec![0, 0, 0, 0, 0, 0])] // regex set with multiple patterns and no matches
+    #[case(vec!["^T", ".GG", "A.+G"], "ATCTACTACG".as_bytes(), bitvec![1, 1, 1, 1, 1, 1, 1, 1, 1, 1])] // regex set where 1/3 patterns have matches
+    #[case(vec!["CT.", ".CT"], "CTACT".as_bytes(), bitvec![1, 1, 1, 1, 1])] // regex set with overlapping matches
+
+    fn test_regex_set_bitvec(
+        #[case] patterns: Vec<&str>,
+        #[case] bases: &[u8],
+        #[case] expected: BitVec,
+    ) {
+        let opts = MatcherOpts {
+            invert_match: false,
+            reverse_complement: false,
+            color: false,
+        };
+
+        let matcher = RegexSetMatcher::new(&patterns, opts);
+
+        let ranges = matcher
+            .regex_matchers
+            .iter()
+            .flat_map(|r| r.regex.find_iter(bases).map(|m| m.range()));
+
+        let result_bitvec = to_bitvec(ranges, bases.len());
+
+        assert_eq!(result_bitvec, expected);
+    }
+
+    // ############################################################################################
+    // Tests validate_fixed_pattern
+    // ############################################################################################
+    #[test]
+    fn test_validate_fixed_pattern_is_ok() {
+        let pattern = "AGTGTGATG";
+        let result = validate_fixed_pattern(&pattern);
+        assert!(result.is_ok())
+    }
+    #[test]
+    fn test_validate_fixed_pattern_error() {
+        let pattern = "A.GTGTGATG";
+        let msg = String::from("Fixed pattern must contain only DNA bases: A .. [.] .. GTGTGATG");
```

Comment:
Makes the expected message clearer.
```suggestion
        let pattern = "AXGTGTGATG";
        let msg = String::from("Fixed pattern must contain only DNA bases: A .. [X] .. GTGTGATG");
```